### DISCUSSION TITLE
Send a file

### DISCUSSION THREAD

**Discussion by miguellz67** [IlyaMatsuev/Apex-GraphQL-Client#36] [Q&A]
I need to send a mutation that takes three variable arguments and one of them is a file, which is mandatory, so I would like to know how I can send this file.

mutation CreateDocumentMutation($document: DocumentInput!, $signers: [SignerInput!]!, $file: Upload!) {createDocument(document: $document, signers: $signers, file: $file) {id name refusable sortable created_at signatures { public_id name email created_at action { name } link { short_link } user { id name email }}}}

**Comment by IlyaMatsuev** [comment]
Hi @miguellz67, I'm afraid I didn't make it possible yet to attach a file to a gql request.
Although, this is a good use case.
I need to investigate how I can do it.
I will update you whenever I have a solution.

For now I would suggest to send a link to the document instead of the file, and then on your graphql server you can download the document by that link.

**Comment by IlyaMatsuev** [comment]
I did some research on sending files via GraphQL API and decided to not implement it as part of this library.
As far as I understand, sharing documents was never a purpose of GraphQL. Usually, a separate endpoint is configured for that.
Some frameworks (like Apollo) implement file uploads but these files are not actually shared via GraphQL format, I assume they're simply using a separate endpoint for that under the hood.

I assume so because I couldn't find anything about file uploads in the GQL specification.
Please let me know if I'm wrong.

As a workaround, you can simply accept a base64 string on your GQL server as a parameter and send your file content in base64 format.
